src/contract_result.ts

```diff
@@ -138,27 +148,22 @@ export class ContractResult<T> {
         const xdr_str = getResult.returnValue?.toXDR('base64');
         return ContractResult.success<T>(hash, resources, parse(xdr_str));
       } else {
-        return ContractResult.error(hash, resources, new Error(`Transaction failed: ${getResult}`));
+        const getResult = response as SorobanRpc.Api.GetFailedTransactionResponse;
+        return ContractResult.error(hash, resources, parseError(getResult.resultXdr.result()));
       }
     }
 
     // otherwise, it returned the result of `sendTransaction`
     if ('errorResultXdr' in response) {
       const sendResult = response as SorobanRpc.Api.SendTransactionResponse;
-      return ContractResult.error(
-        hash,
-        resources,
-        new Error(
-          `Failed to send transaction: ${sendResult.errorResult.toXDR().toString('base64')}`
-        )
-      );
+      return ContractResult.error(hash, resources, parseError(sendResult.errorResult.result()));
     }
 
     // if neither of these are present, something went wrong
     return ContractResult.error(
       hash,
       resources,
-      new Error(`Unable to parse response: ${response.toString()}`)
+      new ContractError(undefined, `Unable to parse response: ${JSON.stringify(response)}`)
```

Comment:
we should return a type for this as well,

`UnknownError` maybe?

we don't need to include the additional "Unable to parse response:" bit then